docs/glossary.md

```diff
@@ -0,0 +1,204 @@
+.
+
+Note that even though commits are stored as snapshots, they are often treated
+as differences between snapshots, namely compared to their parent's snapshot. If
+they have more than one parent, then the difference is computed against the
+result of merging the parents. For example, `jj diff` will show the differences
+introduced by a commit compared to its parent(s), and `jj rebase` will apply
+those changes onto another base commit.
+
+The word "revision" is used as a synonym for "commit".
+
+## Commit ID
+
+A commit ID is a unique identifier for a [commit](#commit). They are 20 bytes
+long when using the Git backend. They are presented in regular hexadecimal
+format at the end of the line in `jj log`, using 12 hexadecimal digits by
+default.
+
+## Co-located repos
+
+When using the Git [backend](#backend) and the backing Git repository's `.git/`
+directory is a sibling of `.jj/`, the we call the repository "co-located".
+
+See [here](git-compatibility.md#co-located-jujutsugit-repos) for details.
+
+## Conflict
+
+Conflicts can occur in many places. The most common type is conflicts in files.
+Those are the conflicts that users coming from other VCSs are usually familiar
+with. You can see them in `jj status` and in `jj log` (the red "conflict"
+label at the end of the line). See [here](conflicts.md) for details.
+
+Conflicts can also occur in [branches](#branch). For example, if you moved a
+branch locally, and it was also moved on the remote, then the branch will be
+in a conflicted state after you pull from the remote.
+See [here](branches.md#conflicts) for details.
+
+Similar to a branch conflict, when a [change](#change) is rewritten locally
+and remotely, for example, then the change will be in a conflicted state. We
+call that a [divergent change](#divergent-change).
+
+## Divergent change
+
+A divergent change is a [change](#change) that has more than one
+[visible commit](#visible-commits).
+
+## Head
+
+A head is a commit with no descendants. The [view](#view) records which
+anonymous heads (heads without a branch pointing to them) are visible at a
+given [operation](#operation). Note that this is quite different from Git's
+[HEAD](https://git-scm.com/book/en/v2/Git-Internals-Git-References#ref_the_ref).
+
+## Operation
+
+A snapshot of the [visible commits](#visible-commits) and [branches](#branches)
+at a given point in time (technically a [view object](#view)), together with
+some metadata. The metadata includes the username, hostname, timestamps, and
+pointers to the operation's parents.
+
+## Operation log
+
+The operation log is the
+[DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph) formed by
+[operation](#operation) objects, much in the same way that commits form a DAG,
+which is sometimes called the "commit history". 
+
+## Repository
+
+Basically everything under `.jj/`, i.e. the full set of [operations](#operation)
+and [commits](#commit).
```

Comment:
I'm again unsure how to improve this (will think about it), but I think it would be nice to more clearly explain the distinction that what an `hg` or `git` user would call a "repository" corresponds to "operation" here, while what we called a repository includes the entire history.

Reply:
I think of all operations as part of the repository. If someone says that their repository is corrupt, I would agree with them even if only some old operation is corrupt. I'd probably also agree even if just the working copy state is corrupt. These kinds of corruption of course have different levels of seriousness, but as I said, I think I'd consider them all "repo corruption".

Reply:
I think this terminology is very reasonable, it is just a little surprising.

I'm still thinking about a better way to phrase it, but it's certainly not a blocker.